docs: adding conventional commits to readme.md (#41138)
owncloud/core: README.md

## Contribution Guidelines
https://owncloud.com/contribute/

## Commit Messages

> To ease bringing commits into context, a CI job takes care that the commit message satisfies a specification for adding human and machine readable meaning to commit messages. For details see: [Conventional Commits](www.conventionalcommits.org/). Note that if conventional commits are not satisfied, CI will not get green. In case, you may need to add an empty commit with a new message.

Author: ```suggestion
To ease bringing commits into context, a CI job takes care that the commit message satisfies a specification for adding human and machine readable meaning to commit messages. For details see: [Conventional Commits](www.conventionalcommits.org/). Note that if conventional commits are not satisfied, CI will not get green. In case, you need to squash and add a commit message satisfying the requirements.
```

> To ease bringing commits into context, a CI job takes care that the commit message satisfies a specification for adding human and machine readable meaning to commit messages. For details see: [Conventional Commits](www.conventionalcommits.org/). Note that if conventional commits are not satisfied, CI will not get green. In case, you need to rewrite the git history to match the requirement..

Reviewer: ```suggestion
To ease bringing commits into context, a CI job check that the commit message satisfies a specification for adding human and machine readable meaning to commit messages. For details see: [Conventional Commits](www.conventionalcommits.org/). Note that if conventional commits are not satisfied, CI will not be green. In this case, you need to rewrite the git commit history to meet the requirement.
```

> You must at minimum provide a `type` + `description` as described in the [Examples](https://www.conventionalcommits.org/en/v1.0.0/#examples) section.

Reviewer: ```suggestion
You must at least provide a `type` + `description` as described in the [Examples](https://www.conventionalcommits.org/en/v1.0.0/#examples) section.
```
